### ISSUE TITLE
Add support fort http request log

### ISSUE THREAD

**Issue by netroby** [spring-projects/spring-boot#4983]
The common httpd server , like apache ,nginx ,could log http request log .

For web applications. we need to log the http request log , for monitor purpose . or analytics purpose.

http://httpd.apache.org/docs/2.2/mod/mod_log_config.html

The apache log format will be more popular today.

would you please add the http log feature for spring boot?

**Comment by vpavic** [comment]
See #2609.

**Comment by wilkinsona** [comment]
We already have support for configuring access logging with Tomcat and Undertow, although it's not very well documented (I've opened #4989 to fix that). It's configured using the following properties:

**Undertow**
```
server.undertow.accesslog.dir= # Undertow access log directory.
server.undertow.accesslog.enabled=false # Enable access log.
server.undertow.accesslog.pattern=common # Format pattern for access logs.
```

**Tomcat**
```
server.tomcat.accesslog.directory=logs # Directory in which log files are created. Can be relative to the tomcat base dir or absolute.
server.tomcat.accesslog.enabled=false # Enable access log.
server.tomcat.accesslog.pattern=common # Format pattern for access logs.
server.tomcat.accesslog.prefix=access_log # Log file name prefix.
server.tomcat.accesslog.suffix=.log # Log file name suffix.
```

The properties are all [listed in the documentation](http://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#common-application-properties).

There's also an open issue for adding support for this with Jetty too: #3018.

In short, I think what you're looking for is already available, or there are already issues tracking an enhancement, so I'm going to close this issue. If you're looking for something else, please comment with some more details and we can re-open the issue.
